R/dataframe-frame.R

```diff
@@ -2216,3 +2216,90 @@ dataframe__describe <- function(
     )
   })
 }
+
+#' Show a dense preview of the DataFrame
+#'
+#' The formatting shows one line per column so that wide DataFrames display
+#' cleanly. Each line shows the column name, the data type, and the first few
+#' values.
+#'
+#' @inheritParams rlang::args_dots_empty
+#' @param max_items_per_column Maximum number of items to show per column.
+#' @param max_colname_length Maximum length of the displayed column names.
+#' Values that exceed this value are truncated with a trailing ellipsis.
+#'
+#' @return Returns a character value (invisibly)
+#'
+#' @examples
+#' df <- as_polars_df(iris)
+#' df$glimpse()
+#'
+#' df$glimpse(max_items_per_column = 3)
+#'
+#' df$glimpse(max_items_per_column = 3, max_colname_length = 3)
+dataframe__glimpse <- function(
+  ...,
+  max_items_per_column = 10,
+  max_colname_length = 50
+) {
```

Comment:
py-polars has an arg `return_as_string` but as in other functions we don't necessarily want this kind of arg. Here I always invisibly return the string.